### PR TITLE
PD-1710 Refresh typography

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/toggle/src/Toggle.tsx [hunk 1]
```diff
@@ -417,6 +417,7 @@ function Toggle({
     >
       <button
         role="switch"
+        type="button"
```
Comment (DesignerDave) [inline packages/toggle/src/Toggle.tsx:420]: Didn't this change already get merged in a different PR?

File: packages/leafygreen-provider/src/TypographyContext.tsx [hunk 2]
```diff
@@ -1,12 +1,13 @@
 import React, { createContext, useContext } from 'react';
 import PropTypes from 'prop-types';
 
+type FontSize = 14 | 16;
```
Comment (DesignerDave) [inline packages/leafygreen-provider/src/TypographyContext.tsx:4]: Should we add a comment to update this later?

**Review by DesignerDave — APPROVED** [review]
Left a couple notes, but otherwise LGTM!

File: packages/toggle/src/Toggle.tsx (hunk 1 above)
Comment (TheSonOfThomp) [reply packages/toggle/src/Toggle.tsx]: I... don't know where this change came from

**Pull request merged**